cosmwasm/packages/wasmi-runtime/src/wasm/contract_validation.rs

```diff
@@ -99,14 +108,39 @@ pub fn validate_contract_key(
     let enclave_key = KEY_MANAGER
         .get_consensus_state_ikm()
         .map_err(|_err| {
-            error!("Error extractling consensus_state_key");
+            error!("Error extracting consensus_state_key");
             false
         })
         .unwrap();
 
     // calculate the authentication_id
     let calculated_authentication_id =
-        generate_contract_id(&enclave_key, &signer_id, &contract_hash);
+        generate_contract_id(&enclave_key, &signer_id, &contract_hash, contract_address);
 
     calculated_authentication_id == expected_authentication_id
 }
+
+pub fn validate_msg(msg: &[u8], contract_code: &[u8]) -> Result<Vec<u8>, EnclaveError> {
+    if msg.len() < HEX_ENCODED_HASH_SIZE {
+        error!("Got message that is missing contract hash for validation");
```

Comment:
I think a better message would be "Malformed message" or something similar? This message could be misleading